pkg/webhook/resources/schedulevmbackup/validator.go

```diff
@@ -0,0 +1,169 @@
+	"github.com/harvester/harvester/pkg/apis/harvesterhci.io/v1beta1"
+	"github.com/harvester/harvester/pkg/controller/master/backup"
+	ctlharvesterv1 "github.com/harvester/harvester/pkg/generated/controllers/harvesterhci.io/v1beta1"
+	"github.com/harvester/harvester/pkg/settings"
+	"github.com/harvester/harvester/pkg/util"
+	werror "github.com/harvester/harvester/pkg/webhook/error"
+	"github.com/harvester/harvester/pkg/webhook/indexeres"
+	"github.com/harvester/harvester/pkg/webhook/types"
+)
+
+const (
+	fieldCron       = "spec.cron"
+	fieldMaxFailure = "spec.maxFailure"
+	fieldSuspend    = "spec.suspend"
+	fieldVMBackup   = "spec.vmbackup"
+)
+
+type scheuldeVMBackupValidator struct {
+	types.DefaultValidator
+	settingCache   ctlharvesterv1.SettingCache
+	secretCache    ctlv1.SecretCache
+	svmbackupCache ctlharvesterv1.ScheduleVMBackupCache
+}
+
+func NewValidator(
+	settingCache ctlharvesterv1.SettingCache,
+	secretCache ctlv1.SecretCache,
+	svmbackupCache ctlharvesterv1.ScheduleVMBackupCache,
+) types.Validator {
+	return &scheuldeVMBackupValidator{
+		settingCache:   settingCache,
+		secretCache:    secretCache,
+		svmbackupCache: svmbackupCache,
+	}
+}
+
+func (v *scheuldeVMBackupValidator) Resource() types.Resource {
+	return types.Resource{
+		Names:      []string{v1beta1.ScheduleVMBackupResourceName},
+		Scope:      admissionregv1.NamespacedScope,
+		APIGroup:   v1beta1.SchemeGroupVersion.Group,
+		APIVersion: v1beta1.SchemeGroupVersion.Version,
+		ObjectType: &v1beta1.ScheduleVMBackup{},
+		OperationTypes: []admissionregv1.OperationType{
+			admissionregv1.Create,
+			admissionregv1.Update,
+			admissionregv1.Delete,
+		},
+	}
+}
+
+func (v *scheuldeVMBackupValidator) checkTargetHealth() error {
+	targetSetting, err := v.settingCache.Get(settings.BackupTargetSettingName)
+	if err != nil {
+		return err
+	}
+
+	if targetSetting.Value == "" {
+		return fmt.Errorf("setting %s is empty", settings.BackupTargetSettingName)
+	}
+
+	target, err := settings.DecodeBackupTarget(targetSetting.Value)
+	if err != nil {
+		return fmt.Errorf("can't decode setting %s value %s, error: %w", settings.BackupTargetSettingName, targetSetting.Value, err)
+	}
+
+	if target.IsDefaultBackupTarget() {
+		return fmt.Errorf("setting %s is not set", settings.BackupTargetSettingName)
+	}
+
+	if target.Type == settings.S3BackupType {
+		secret, err := v.secretCache.Get(util.LonghornSystemNamespaceName, util.BackupTargetSecretName)
+		if err != nil {
+			return fmt.Errorf("can't get backup target secret: %s/%s, error: %w",
+				util.LonghornSystemNamespaceName, util.BackupTargetSecretName, err)
+		}
+		os.Setenv(backup.AWSAccessKey, string(secret.Data[backup.AWSAccessKey]))
+		os.Setenv(backup.AWSSecretKey, string(secret.Data[backup.AWSSecretKey]))
+		os.Setenv(backup.AWSEndpoints, string(secret.Data[backup.AWSEndpoints]))
+		os.Setenv(backup.AWSCERT, string(secret.Data[backup.AWSCERT]))
+	}
+
+	_, err = backupstore.GetBackupStoreDriver(backup.ConstructEndpoint(target))
```

Comment:
We can use `util.GetBackupStoreDriver`, so we don't need to handle s3 backup type here.

https://github.com/harvester/harvester/blob/b28f8d79f8680f29bd9c5402c71c566837788c7a/pkg/util/backup.go#L26C27-L44

Reply:
Updated per suggestion, thanks.